### DISCUSSION TITLE
Reading values from SGP30

### DISCUSSION THREAD

**Discussion by mahfoozm** [mattjlewis/diozero#92] [Q&A] [answered]
Is there a way to read CO2 values from an SGP30 (connected to an Arduino Uno)? I tried modifying the SGP30 example to configure it to work for my Arduino, but had no luck. Thanks! 

(FYI, this is with the Arduino configured with StandardFirmata)

**Comment by mattjlewis** [comment, marked as answer]
Hi, unfortunately this isn't possible via Firmata - I had to resort to using raw I2C read/write instructions to get this device to work, these aren't supported with the Firmata protocol.
[This is](https://github.com/mattjlewis/diozero/blob/main/diozero-core/src/main/java/com/diozero/devices/SGP30.java#L252) the relevant line of code where I can see I added a comment regarding this.

**Reply by mahfoozm** [reply]
Ah, that's a shame. Thank you though!
